### PR TITLE
Error out early if system does not support pre-copy  checkpointing

### PR DESCRIPTION
**Pull request by adrianreber**
CRIU's pre-copy migration support relies on the soft dirty page tracking in the Linux kernel:

 https://www.kernel.org/doc/Documentation/vm/soft-dirty.txt

This functionality is not implemented for all architectures and it can also be turned off in the kernel.

CRIU can check if the combination of architecture/kernel/CRIU supports the soft dirty page tracking and exports this feature checking functionality in go-criu.

This commit adds an early check if the user selects pre-copy checkpointing to error out if the system does not support it.

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

File: docs/source/markdown/podman-container-checkpoint.1.md [hunk 1]
```diff
@@ -126,6 +133,11 @@ Check out the *container* with previous criu image files in pre-dump. It only wo
 The default is **false**.\
 *IMPORTANT: This OPTION is not available with __--pre-checkpoint__*.
 
+This option requires that the option __--pre-checkpoint__ has been used before on the
+same container. Without an existing pre-checkpoint this option will fail.
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-container-checkpoint.1.md:137]:
```suggestion
same container. Without an existing pre-checkpoint, this option will fail.
```

File: docs/source/markdown/podman-container-checkpoint.1.md [hunk 2]
```diff
@@ -70,6 +70,13 @@ Dump the *container's* memory information only, leaving the *container* running.
 operations will supersede prior dumps. It only works on `runc 1.0-rc3` or `higher`.\
 The default is **false**.
 
+The functionality to only checkpoint the memory of the container and in a second
+checkpoint only write out the memory pages which have changed since the first
+checkpoint relies on the Linux kernel's soft-dirty bit which is not available
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-container-checkpoint.1.md:75]:
```suggestion
checkpoint relies on the Linux kernel's soft-dirty bit, which is not available
```

**Comment by TomSweeneyRedHat** [comment]
Tests aren't happy.  Couple doc nits, otherwise LGTM

File: docs/source/markdown/podman-container-checkpoint.1.md (hunk 1 above)
Comment (adrianreber) [reply docs/source/markdown/podman-container-checkpoint.1.md]: Thanks. Fixed.

File: docs/source/markdown/podman-container-checkpoint.1.md (hunk 2 above)
Comment (adrianreber) [reply docs/source/markdown/podman-container-checkpoint.1.md]: Thanks. Fixed.

**Comment by mheon** [comment]
I think this was Quay hiccuping - restarted

File: pkg/criu/criu.go [hunk 3]
```diff
@@ -21,3 +26,16 @@ func CheckForCriu(version int) bool {
...
+	features, err := criu.MakeCriu().FeatureCheck(
+		&rpc.CriuFeatures{
+			MemTrack: proto.Bool(true),
+		},
+	)
+	if err != nil {
+		return false
+	}
+
+	return *features.MemTrack
```
Comment (mheon) [inline pkg/criu/criu.go:40]: Can you do a nil-check on this before returning? A null return may be unlikely, but not segfaulting is important.

**Comment by mheon** [comment]
One nit otherwise LGTM

File: pkg/criu/criu.go (hunk 3 above)
Comment (adrianreber) [reply pkg/criu/criu.go:44]: done

**Comment by rhatdan** [comment]
@adrianreber Needs a rebase.

**Comment by adrianreber** [comment]
Rebased

**Pull request merged**